python/tests/test_coffeeshop.py

```diff
@@ -0,0 +1,9 @@
+from typechat import python_type_to_typescript_schema
+from tests.coffeeshop import Cart
+
+def test_python_to_typescript_schema():
+    result = python_type_to_typescript_schema(Cart)
+    assert result is not None
+    assert result.typescript_type_reference == "Cart"
+    assert result.typescript_schema_str is not None
+    assert not result.errors
```

Comment:
So I'm not entirely familiar with it, but if we add syrupy, it looks like you can add a parameter called `snapshot` and you can just write

```suggestion
def test_coffeeshop(snapshot):
    result = python_type_to_typescript_schema(Cart)
    assert result == snapshot
```

Let me know how that works

Reply:
Worked out really well. Except it looks like the function python_type_to_typescript_schema produced the python to typescript conversions in an order that could change. Look at the outputs:

  **assert [+ received] == [- snapshot]**
`E         - TypeScriptSchemaConversionResult(typescript_schema_str='interface Cart {\n    type: "Cart";\n    items: Array<LineItem | UnknownText>;\n}\n\ninterface LineItem {\n    type: "LineItem";\n    product: BakeryProduct | LatteDrink | CoffeeDrink | EspressoDrink | UnknownText;\n    quantity: number;\n}\n\ninterface LatteDrink {\n    type: "LatteDrink";\n    name: "cappuccino" | "flat white" | "latte" | "latte macchiato" | "mocha" | "chai latte";\n    temperature?: "hot" | "extra hot" | "warm" | "iced";\n    // The default is \'grande\'\n    size?: "short" | "tall" | "grande" | "venti";\n    options?: Array<Creamer | Sweetener | Syrup | Topping | Caffeine | LattePreparation>;\n}\n\ninterface CoffeeDrink {\n    type: "CoffeeDrink";\n    name: "americano" | "coffee";\n    temperature?: "hot" | "extra hot" | "warm" | "iced";\n    // The default is \'grande\'\n    size?: "short" | "tall" | "grande" | "venti";\n    options?: Array<Creamer | Sweetener | Syrup | Topping | Caffeine | LattePreparation>;\n}\n\ninterface BakeryProduct {\n    type: "BakeryProduct";\n    name: "apple bran muffin" | "blueberry muffin" | "lemon poppyseed muffin" | "bagel";\n    options?: Array<BakeryOption | BakeryPreparation>;\n}\n\ninterface Topping {\n    type: "Topping";\n    name: "cinnamon" | "foam" | "ice" | "nutmeg" | "w...`


`E         + TypeScriptSchemaConversionResult(typescript_schema_str='interface Cart {\n    type: "Cart";\n    items: Array<LineItem | UnknownText>;\n}\n\ninterface LineItem {\n    type: "LineItem";\n    product: BakeryProduct | LatteDrink | CoffeeDrink | EspressoDrink | UnknownText;\n    quantity: number;\n}\n\ninterface BakeryProduct {\n    type: "BakeryProduct";\n    name: "apple bran muffin" | "blueberry muffin" | "lemon poppyseed muffin" | "bagel";\n    options?: Array<BakeryOption | BakeryPreparation>;\n}\n\ninterface BakeryPreparation {\n    type: "BakeryPreparation";\n    name: "warmed" | "cut in half";\n}\n\n// Represents any text that could not be understood.\ninterface UnknownText {\n    type: "UnknownText";\n    // The text that wasn\'t understood\n    text: string;\n}\n\ninterface CoffeeDrink {\n    type: "CoffeeDrink";\n    name: "americano" | "coffee";\n    temperature?: "hot" | "extra hot" | "warm" | "iced";\n    // The default is \'grande\'\n    size?: "short" | "tall" | "grande" | "venti";\n    options?: Array<Creamer | Sweetener | Syrup | Topping | Caffeine | LattePreparation>;\n}\n\ninterface LattePreparation {\n    type: "LattePreparation";\n    name: "for here cup" | "lid" | "with room" | "to go" | "dry" | "wet";\n}\n\ninterface LatteDrink {\n    type: "LatteDrink";\n    name: "cap...`

Reply:
As a result the snapshot tests fail. Once we can make the output more deterministic this will work out great.

Reply:
I'm extremely surprised - you're saying across runs the output changes?

Reply:
Duh, sets and dicts don't specify an iteration order in Python - they are guaranteed to have insertion order in JavaScript.

Reply:
I'll fix that by Monday.

Reply:
No worries. The other is test failing was the hello world one but I guess all of them might have the same issue. We need to maintain the order to be able to run the snapshot tests.

Reply:
#192 should fix the instances in the Py->TS translation where order matters on sets and dicts